SDL/Event.cu

```diff
@@ -640,57 +647,82 @@ public:
+    -> void
     {
-        float ihit_x = hitsInGPU->xs[ihit];
-        float ihit_y = hitsInGPU->ys[ihit];
-        float ihit_z = hitsInGPU->zs[ihit];
-        int iDetId = hitsInGPU->detid[ihit];
 
-        hitsInGPU->rts[ihit] = sqrt(ihit_x*ihit_x + ihit_y*ihit_y);
-        hitsInGPU->phis[ihit] = SDL::phi(ihit_x,ihit_y,ihit_z);
-        hitsInGPU->etas[ihit] = ((ihit_z>0)-(ihit_z<0))*acosh(sqrt(ihit_x*ihit_x+ihit_y*ihit_y+ihit_z*ihit_z)/hitsInGPU->rts[ihit]);
+        TIdx const gridThreadIdx(alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u]);
+        TIdx const threadElemExtent(alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u]);
+        TIdx const threadFirstElemIdx(gridThreadIdx * threadElemExtent);
 
-        int found_index = binary_search(modulesInGPU->mapdetId, iDetId, nModules);
-        uint16_t lastModuleIndex = modulesInGPU->mapIdx[found_index];
+        if(threadFirstElemIdx < nHits)
+        {
+            TIdx const threadLastElemIdx(threadFirstElemIdx+threadElemExtent);
+            TIdx const threadLastElemIdxClipped((nHits > threadLastElemIdx) ? threadLastElemIdx : nHits);
 
-        hitsInGPU->moduleIndices[ihit] = lastModuleIndex;
+            for(TIdx ihit(threadFirstElemIdx); ihit<threadLastElemIdxClipped; ++ihit)
+            {
+                // Is there a better way to do this?
+                int ihit_int = ihit;
 
-        if(modulesInGPU->subdets[lastModuleIndex] == Endcap && modulesInGPU->moduleType[lastModuleIndex] == TwoS)
-        {
-            int found_index = binary_search(geoMapDetId, iDetId, nEndCapMap);
-            float phi = 0;
-            // Unclear why these are not in map, but CPU map returns phi = 0 for all exceptions.
-            if (found_index != -1)
-                phi = geoMapPhi[found_index];
-            float cos_phi = cosf(phi);
-            hitsInGPU->highEdgeXs[ihit] = ihit_x + 2.5f * cos_phi;
-            hitsInGPU->lowEdgeXs[ihit] = ihit_x - 2.5f * cos_phi;
-            float sin_phi = sinf(phi);
-            hitsInGPU->highEdgeYs[ihit] = ihit_y + 2.5f * sin_phi;
-            hitsInGPU->lowEdgeYs[ihit] = ihit_y - 2.5f * sin_phi;
-        }
-        // Need to set initial value if index hasn't been seen before.
-        int old = atomicCAS(&(hitsInGPU->hitRanges[lastModuleIndex * 2]), -1, ihit);
-        // For subsequent visits, stores the min value.
-        if (old != -1)
-            atomicMin(&hitsInGPU->hitRanges[lastModuleIndex * 2], ihit);
+                float ihit_x = hitsInGPU->xs[ihit];
+                float ihit_y = hitsInGPU->ys[ihit];
+                float ihit_z = hitsInGPU->zs[ihit];
+                int iDetId = hitsInGPU->detid[ihit];
+        
+                hitsInGPU->rts[ihit] = alpaka::math::sqrt(acc, ihit_x*ihit_x + ihit_y*ihit_y);
+                // This needs to get moved over after all kernels that use it have been moved.
+                hitsInGPU->phis[ihit] = SDL::phi(ihit_x,ihit_y,ihit_z);
```

Comment:
Probably unrelated to this PR but why do we need `z` in the calculation of `phi`?

Reply:
I don't think it's necessary, but this is probably something that should be fixed on master.